# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/base/TextLeafRange.h
# Commit: 547c6d91f7e9
# Full Hash: 547c6d91f7e99909a4ec3f96327251641705d217
# Author: James Teh <jteh@mozilla.com>
# Date: 2025-05-07 04:19:47
# Regressor Bug: 1869966
# File Overlap Count: 1
# Description:
#   Bug 1869966 part 3: When calculating a DOM point from a TextLeafPoint for a non-text node, use the child index within the parent, not an offset within the node itself. r=eeejay
#   
#   For text nodes, DOM points use the character offset within the node itself, just like accessibility does.
#   For non-text nodes, DOM points use the child index within the parent to specify the position before or after the node.
#   Previously, TextLeafPoint::ToDOMPoint always used an offset within the node itself.
# ==============================================================================

diff -r 16652a9ee9f7 -r 547c6d91f7e9 accessible/base/TextLeafRange.h
--- a/accessible/base/TextLeafRange.h	Tue May 06 22:13:58 2025 +0000
+++ b/accessible/base/TextLeafRange.h	Tue May 06 22:13:58 2025 +0000
@@ -188,7 +188,7 @@
   /**
    * Translate given TextLeafPoint into a DOM point.
    */
-  MOZ_CAN_RUN_SCRIPT std::pair<nsIContent*, int32_t> ToDOMPoint(
+  MOZ_CAN_RUN_SCRIPT std::pair<nsIContent*, uint32_t> ToDOMPoint(
       bool aIncludeGenerated = true) const;
 
  private: